spec/v1.0-draft/field_types.md

```diff
@@ -46,6 +46,55 @@ _DigestSet (object)_
 > -   `{"sha256": "abcd"}` does not match `{"sha256": "fedb", "sha512": "abcd"}`
 > -   `{"somecoolhash": "abcd"}` uses a non-predefined algorithm
 
+<a id="Reference"></a>
+_Reference (object)_
+
+> A size-efficient representation of any artifact, metadata or attestation,
+> referenced by another in-toto attestation.
+> This type is designed to be backwards-compatible with the in-toto
+> attestation [Statement v0.1][] as well as the [SLSA ArtifactReference][].
+>
+> Schema:
+>
+>```jsonc
+> {
+>   "name": "<NAME>",
+>   "digest": { "<ALGORITHM>": "<HEX VALUE>", ... },
+>   "uri": "<RESOURCE URI>",
+>   "downloadURI": "<RESOURCE URI>", // optional
+>   "mimeType": "<MIME TYPE>" // optional
+> }
+>```
+>
+> Fields:
+>
+> -   `name` (_string, required_): Human-readable identifier to distinguish
+>     the referenced object locally. The semantics are up to the producer and
+>     consumer.
+> -   `digest` (_[DigestSet], cond. required_): The producer and
+>     consumer must agree on acceptable algorithms. This field MUST be
+>     specified if the `uri` field is not present.
+> -   `uri` (_[ResourceURI], cond. required_): URI for the referenced
```

Comment:
What's the difference between `downloadUri` and `uri`?  Do we need both?  If so, can they both be optional?

Reply:
These two fields are mirroring the SLSA v1 [ArtifactReference](https://github.com/slsa-framework/slsa/blob/main/docs/provenance/v1/index.md#artifactreference) field type, where `uri` is more of a type descriptor than a pointer/hint to the location of the artifact. I wanted to make sure this field type is compatible with SLSA Provenance, but I also agree that `downloadURI` should be enough, given the `mimeType` field also exists.

Reply:
Oooh, I see.  I wonder if we should consider mirroring that more closely?

E.g. 'downloadLocation' and 'mediaType' ?  Maybe even mirror their language?

I'm not sure about renaming 'name' to 'localName' since it would break compatibility with version 0.1, which just makes adoption of marking this 1.0 slightly harder.

I could see where SLSA could use this type instead of defining their own.  In which case we could even consider calling this 'ArtifactReference'?  Perhaps that's too ambitious?

I'd be open for either, I think my primary motivation is that it would be great if attestation v1 was ready for SLSA v1.  So if we can do both then great?

Reply:
+1 to having attestation v1 ready for SLSA v1.

I'm generally fine renaming the fields to `mediaType` and `downloadLocation`. Agreed that switching to `localName` would be a little awkward. FWIW, the current SLSA `ArtifactReference` has an RFC on whether to keep both the `uri` and `downloadLocation` fields, so maybe we can also bring this discussion back to the SLSA community. Either way, I agree that it would be ideal if this reference type could just be used in SLSA instead of defining its own, very similar, type.

My main hesitation to rename the type to `ArtifactReference` is that we want to be able to point to "non-artifacts" like SBOM, other in-toto attestations and such, so I'd be worried that the term "artifact" would be misleading. Another option is going with the more general `ObjectReference` (a la SCAI) naming instead. But I could go either way, especially if we say that attestations/SBOM/etc are in fact artifacts, too. I guess we just need to see what folks prefer.